spring-cloud-stream-binders/spring-cloud-stream-binder-redis/src/main/java/org/springframework/cloud/stream/binder/redis/RedisMessageChannelBinder.java

```diff
@@ -300,55 +243,39 @@ public Boolean recover(RetryContext context) throws Exception {
 	}
 
 	@Override
-	public void bindProducer(final String name, MessageChannel moduleOutputChannel,
-			Properties properties) {
-		Assert.isInstanceOf(SubscribableChannel.class, moduleOutputChannel);
-		if (name.startsWith(P2P_NAMED_CHANNEL_TYPE_PREFIX)) {
-			validateProducerProperties(name, properties, SUPPORTED_NAMED_PRODUCER_PROPERTIES);
-		}
-		else {
-			validateProducerProperties(name, properties, SUPPORTED_PRODUCER_PROPERTIES);
-		}
-		RedisPropertiesAccessor accessor = new RedisPropertiesAccessor(properties);
-		if (!bindNewProducerDirectlyIfPossible(name, (SubscribableChannel) moduleOutputChannel, accessor)) {
-			String partitionKeyExtractorClass = accessor.getPartitionKeyExtractorClass();
-			Expression partitionKeyExpression = accessor.getPartitionKeyExpression();
-			RedisQueueOutboundChannelAdapter queue;
-			String queueName = "queue." + name;
-			if (partitionKeyExpression == null && !StringUtils.hasText(partitionKeyExtractorClass)) {
-				queue = new RedisQueueOutboundChannelAdapter(queueName, this.connectionFactory);
-			}
-			else {
-				queue = new RedisQueueOutboundChannelAdapter(
-						parser.parseExpression(buildPartitionRoutingExpression(queueName)), this.connectionFactory);
-			}
-			queue.setIntegrationEvaluationContext(this.evaluationContext);
-			queue.setBeanFactory(this.getBeanFactory());
-			queue.afterPropertiesSet();
-			doRegisterProducer(name, moduleOutputChannel, queue, accessor);
-		}
+	public void unbindConsumers(String name, String group) {
+		super.unbindConsumers(name, group);
+		this.redisOperations.boundSetOps(CONSUMER_GROUPS_KEY_PREFIX + name).remove(group);
```

Comment:
I think using a ZSET might work - with the count as the value. Still a "set" so that's good.